pkg/driver/controller.go

```diff
@@ -0,0 +1,326 @@
+			return nil, status.Errorf(codes.InvalidArgument, "File System does not exist: %v", err)
+		}
+		return nil, status.Errorf(codes.Internal, "Failed to fetch File System info: %v", err)
+	}
+	gid, err := d.gidAllocator.getNextGid(accessPointsOptions.FileSystemId, gidMin, gidMax)
+	if err != nil {
+		return nil, err
+	}
+
+	gidStr := strconv.Itoa(gid)
+	rootDirName := RootDirPrefix + "-" + gidStr + "-" + uuid.New().String()
+	rootDir := basePath + "/" + rootDirName
+
+	accessPointsOptions.Gid = int64(gid)
+	accessPointsOptions.Uid = int64(gid)
+	accessPointsOptions.DirectoryPath = rootDir
+
+	accessPointId, err := d.cloud.CreateAccessPoint(ctx, volName, accessPointsOptions)
+	if err != nil {
+		d.gidAllocator.releaseGid(accessPointsOptions.FileSystemId, gid)
+		if err == cloud.ErrAccessDenied {
+			return nil, status.Errorf(codes.Unauthenticated, "Access Denied. Please ensure you have the right AWS permissions: %v", err)
+		}
+		if err == cloud.ErrAlreadyExists {
+			return nil, status.Errorf(codes.AlreadyExists, "Access Point already exists")
+		}
+		return nil, status.Errorf(codes.Internal, "Failed to create Access point in File System %v : %v", accessPointsOptions.FileSystemId, err)
+	}
+
+	if accessPointsOptions.DeleteContent {
+		volContext[DeleteContent] = strconv.FormatBool(accessPointsOptions.DeleteContent)
+	}
+
+	return &csi.CreateVolumeResponse{
+		Volume: &csi.Volume{
+			CapacityBytes: volSize,
+			VolumeId:      accessPointsOptions.FileSystemId + "::" + accessPointId.AccessPointId,
+			VolumeContext: volContext,
+		},
+	}, nil
+}
+
+func (d *Driver) DeleteVolume(ctx context.Context, req *csi.DeleteVolumeRequest) (*csi.DeleteVolumeResponse, error) {
+	klog.V(4).Infof("DeleteVolume: called with args %+v", *req)
+	volId := req.GetVolumeId()
+	if volId == "" {
+		return nil, status.Error(codes.InvalidArgument, "Volume ID not provided")
+	}
+
+	_, _, accessPointId, err := parseVolumeId(volId)
+	if err != nil {
+		if err == cloud.ErrAccessDenied {
+			return nil, status.Errorf(codes.Unauthenticated, "Access Denied. Please ensure you have the right AWS permissions: %v", err)
+		}
+		//Returning success for an invalid volume ID. See here - https://github.com/kubernetes-csi/csi-test/blame/5deb83d58fea909b2895731d43e32400380aae3c/pkg/sanity/controller.go#L733
+		klog.V(5).Infof("DeleteVolume: Failed to parse volumeID: %v, err: %v, returning success", volId, err)
+		return &csi.DeleteVolumeResponse{}, nil
+	}
+
+	//TODO: Add Delete File System when FS provisioning is implemented
+	if accessPointId != "" {
+		if err = d.cloud.DeleteAccessPoint(ctx, accessPointId); err != nil {
+			if err == cloud.ErrNotFound {
+				klog.V(5).Infof("DeleteVolume: Access Point not found, returning success")
+				return &csi.DeleteVolumeResponse{}, nil
+			}
+			return nil, status.Errorf(codes.Internal, "Failed to Delete volume %v: %v", volId, err)
+		}
+	} else {
+		return nil, status.Errorf(codes.Internal, "Failed to find access point for volume: %v", volId)
+	}
```

Comment:
Is this an internal failure (assuming `codes.Internal` means it's 500x)?

Reply:
This will be an internal error from AWS, so I guess `codes.Internal`?
Or should we encapsulate all unexpected errors under `codes.Unknown`?

Reply:
Why is failing to find access point a 500x not 400x?

Reply:
Sorry, I mistook this to be inside the `if` loop above. I'll update this to `codes.NotFound`